stock_picking_show_return/models/stock_picking.py

```diff
@@ -10,9 +11,28 @@ class StockPicking(models.Model):
     returned_ids = fields.Many2many(
         comodel_name="stock.picking", compute="_compute_returned_ids",
         string="Returned pickings")
+    origin_returned_id = fields.Many2one(
+        comodel_name="stock.picking",
+        compute="_compute_origin_returned_id",
+        string="Origin picking")
 
     @api.multi
     def _compute_returned_ids(self):
         for picking in self:
             picking.returned_ids = picking.mapped(
                 'move_lines.returned_move_ids.picking_id')
+
+    @api.multi
+    def _compute_origin_returned_id(self):
+        """
+        Get origin piking from this returned picking. Only one origin is
+        possible.
+        """
+        for picking in self:
+            picking.origin_returned_id = picking.mapped(
+                'move_lines.origin_returned_move_id.picking_id')
```

Comment:
```suggestion
                'move_lines.origin_returned_move_id.picking_id')[:1]
```
for being sure only one is returned in case of inconsistent data.